### PR TITLE
improve required_env docs

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/compute 

Ankit found them on his own, but they were not very clear ...

### CODE REVIEW

**Review by hgokavarapuz — APPROVED** [review]

File: plugins/kubernetes/README.md [hunk 1]
```diff
@@ -67,8 +67,14 @@ environment variables, secrets, etc and then sends them to the Kubernetes API.
 Multiple roles will likely be similar, but have different commands or liveness probes.
 ([kucodiff](https://github.com/grosser/kucodiff) can be used to make sure they stay in sync).
 
-Samson can be instructed to validate certain environment variables are present by adding a `samson/required_env`
-annotation (space separated).
+Validate required environment variables are set by adding `metadata.annotations.samson/required_env` (space separated).
```
Comment (adammw) [inline plugins/kubernetes/README.md]: adding `samson/required_env` annotation makes more sense to me.

**Review by adammw — APPROVED** [review]

File: plugins/kubernetes/README.md (hunk 1 above)
Comment (grosser) [reply plugins/kubernetes/README.md]: I'm trying to keep this file consistent so everything has a full path and I notice when I forget to mention exactly where things go more easily :)

**Comment by adammw** [comment]
Worth mentioning it should be set on the metadata of the pod (template), not the deployment?

**Comment by grosser** [comment]
yeah :/

**Pull request merged**